### PR TITLE
feat: search for and view other users' profile

### PR DESCRIPTION
**Pull request by c25vdw**

### CODE REVIEW

**Comment by c25vdw** [comment]
<details><summary>demo</summary>
<p></p>
</details>

File: app/src/main/java/com/cmput301f20t21/bookfriends/ui/profile/ProfileViewModel.java [hunk 1]
```diff
@@ -1,8 +1,78 @@
...
+            if (usernameTask.isSuccessful()) {
+                if (!usernameTask.getResult().isEmpty()) {
+                    ArrayList<User> users = new ArrayList<>();
+                    for (QueryDocumentSnapshot document: usernameTask.getResult()) {
+                        users.add(new User(
+                                document.getId(),
+                                document.get("username").toString(),
+                                "",
+                                document.get("email").toString(),
+                                ""
```
Comment (kdvuong) [inline app/src/main/java/com/cmput301f20t21/bookfriends/ui/profile/ProfileViewModel.java:42]: In the requirement, we only need to display user name and email. I'm thinking of removing phone number and display name to simplify everything.

Comment (c25vdw) [reply app/src/main/java/com/cmput301f20t21/bookfriends/ui/profile/ProfileViewModel.java:42]: sounds good to me. should we do that in another PR?

**Review by zhpeng811 — APPROVED** [review]
lgtm

**Review by kdvuong — APPROVED** [review]
lgtm

**Pull request merged**